ext/json/json.c

```diff
@@ -31,6 +31,14 @@
 #include "php_json.h"
 #include <zend_exceptions.h>
 
+#include <float.h>
+#if defined(DBL_MANT_DIG) && defined(DBL_MIN_EXP)
+#define NUM_BUF_SIZE (3 + DBL_MANT_DIG - DBL_MIN_EXP)
+#else
+#define NUM_BUF_SIZE 1080
```

Comment:
Btw, the constants are defined on windows, linux and mac. So just in rare cases it will fallback to the hardcoded value.